part2/stages/TPM-Check

```diff
@@ -25,19 +25,29 @@ if [ "${MEASURE_LAUNCH}" = "false" ]; then
 fi
 
 TPM_FUNCS=/usr/lib/tpm-scripts/tpm-functions
+KEY_FUNCS=/usr/lib/openxt/key-functions
 if [ ! -f ${TPM_FUNCS} ]; then
     echo "No tpm-function library found at ${TPM_FUNCS}.  ABORT" >&2
     exit ${Abort}
 else
     . ${TPM_FUNCS}
 fi
 
-MLE_FUNCS=/usr/lib/openxt/ml-functions
-if [ ! -f ${MLE_FUNCS} ]; then
-    echo "No ml-function library found at ${MLE_FUNCS}.  ABORT" >&2
+if [ ! -f ${KEY_FUNCS} ]; then
+    echo "No key-functions library found at ${KEY_FUNCS}.  ABORT" >&2
     exit ${Abort}
 else
-    . ${MLE_FUNCS}
+    . ${KEY_FUNCS}
+fi
+
+if [ ! -d /sys/firmware/efi/efivars ]; then
```

Comment:
You have a few places where you check this directory.  Would it make sense to set a variable for EFI vs. BIOS or SRTM vs. DRTM installation?

Reply:
Unfortunately the case is not that clear cut, mostly due to the fact that dom0 is still 32-bit. It is entirely possible to have a UEFI install and have some of these scripts run (during an upgrade) without actually having a functional efivarfs mounted.